src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -210,6 +215,22 @@ private async Task InitializeAsync()
             hr = _vsSolution.AdviseSolutionEvents(this, out _solutionEventsCookie);
             ErrorHandler.ThrowOnFailure(hr);
 
+            string solutionFileName;
+            string solutionDirectory;
+            ErrorHandler.ThrowOnFailure(_vsSolution.GetSolutionInfo(out solutionDirectory, out solutionFileName, out _));
```

Comment:
Can GetSolutionInfo ever fail? 

Should we just queitly assume the solution isn't open if it fails?

Reply:
I was looking at the implementation (yay, benefit of being Microsoft internal), and it's kind of complicated, but `GetSolutionInfo` just gets properties from the solution, so I just changed it to use fewer indirection. `GetProperty` will return an error when the solution is closed, but like you said, we use that as the trigger to figure out it's closed, rather than thorwing.